api/metadata/serializers.py

```diff
@@ -0,0 +1,133 @@
+from django.contrib.contenttypes.models import ContentType
+from django.db.models import Q
+from rest_framework import serializers
+
+from .models import (
+    FIELD_VALUE_MAX_LENGTH,
+    FieldType,
+    Metadata,
+    MetadataField,
+    MetadataModelField,
+)
+
+
+class MetadataFieldSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = MetadataField
+        fields = ("id", "name", "type", "description", "organisation")
+
+
+class MetaDataModelFieldSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = MetadataModelField
+        fields = ("id", "field", "is_required")
+
+
+class FieldDataField(serializers.Field):
+    """
+    Custom field to handle the different types of data supported by the metadata field
+    """
+
+    def to_internal_value(self, data):
```

Comment:
Do we need to do this? Shouldn't we just get the client to send up the type and we can always just store the data as a string?

Reply:
But we already have the expected type for this, right? this is used to verify it's type

Reply:
Hmm... I think we should just get the FE to send a string (so we don't have to hack the API docs) and then do something more like: 

```python
casting_functions = {"int": int, "str": str, ...}

try:
    casting_function = casting_functions.get(field_type)  # we might need to use context to get the field type here? 
    casting_function(data)
except ValueError:
    raise ValidationError(...)

if len(str(data)) > FIELD_VALUE_MAX_LENGTH:
    raise ValidationError(...)

return data
```